javascript/general_style_guide.md

```diff
@@ -0,0 +1,13 @@
+# Javascript Style Guide
+
+Our Javascript Code Style is determined by `prettier`. Please make sure files have
+passed through `prettier` before committing them to the repo.
```

Comment:
I normally align this with `make run-prettier`, but would it make sense to write here simple instructions how to set it up to VS.Code which I think many use for front-end stuff? I noticed there are tons of plugins for this and I'm not sure if the default style in JS world is the same like `gofmt` in Go world. Or does this make sense at all?

Reply:
Javascript is no where near as standardised when it comes to tooling.

There is no 'standard' type of thing, though lots of people have been accepting 'prettier' as the formatter these days. The way to invoke it (whether by `make` or some other task runner) is not really standardised.

Also lots of different editors. I for example use Sublime still... though I might switch to VS.Code.

Reply:
I tried for ten minutes and gave up to set it up in Atom.